src/Fission.re

```diff
@@ -1,87 +1,85 @@
+  |> Js.Promise.catch(Js.Promise.resolve);
+
+let octetHeader = Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+
+module type URL = {
+  let baseUrl: string;
+};
+
+module type Auth = {
+  let username: string;
+  let password: string;
 };
 
-let octetHeader =
-  Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+module Simple = (Base: URL) => {
+  let ipfsURL = Base.baseUrl ++ "/ipfs";
+  let cidsURL = ipfsURL ++ "/cids"
+  let url     = cid => ipfsURL ++ cid;
 
-let convAuth = (auth: auth) => {
-  "username": auth.username,
-  "password": auth.password,
+  let content = cid =>
+    cid
+    -> url
+    -> Axios.get
+    -> await;
 };
 
-let url = (baseURL: string, cid: cid) => baseURL ++ "/ipfs/" ++ cid;
-
-let content = (cid: cid) =>
-  Js.Promise.(
-    Axios.get(url(baseURL, cid))
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-let list = (auth: auth) =>
-  Js.Promise.(
-    Axios.getc(
-      baseURL ++ "/ipfs/cids",
-      Axios.makeConfig(~auth=convAuth(auth), ()),
-    )
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-let add = (content: 'a, auth: auth) =>
-  Js.Promise.(
-    Axios.postDatac(
-      baseURL ++ "/ipfs/",
-      content,
-      Axios.makeConfig(~auth=convAuth(auth), ~headers=octetHeader, ()),
-    )
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-let addStr = (_str: string, auth: auth) =>
-  Js.Promise.(
-    Axios.postDatac(
-      baseURL ++ "/ipfs/",
-      [%bs.raw {|str|}],
-      Axios.makeConfig(~auth=convAuth(auth), ~headers=octetHeader, ()),
-    )
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-let pin = (cid: cid, auth: auth) =>
-  Js.Promise.(
-    Axios.putDatac(
-      url(baseURL, cid),
-      [%bs.raw {|{}|}],
-      Axios.makeConfig(~auth=convAuth(auth), ()),
-    )
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-let remove = (cid: cid, auth: auth) =>
-  Js.Promise.(
-    Axios.deletec(
-      url(baseURL, cid),
-      Axios.makeConfig(~auth=convAuth(auth), ()),
-    )
-    |> then_(response => resolve(response##data))
-    |> catch(error => resolve(error))
-  );
-
-// testing purposes, delete later
-let user = {username: env_username, password: env_password};
-Js.Promise.(
-  remove("QmQbPPkak9itW3v8WSohtonCBiJcrnAUhrSW1TGPnmWe3f", user)
-  |> then_(result => resolve(Js.Console.log(result)))
-  |> catch(error => resolve(Js.Console.log(error)))
-);
\ No newline at end of file
+module User = (Base: URL, Auth: Auth) => {
+  include Simple(Base);
+
+  let auth = {
+    "username": Auth.username,
+    "password": Auth.password,
+  };
+
+  let octetConfig = Axios.makeConfig(~auth, ~headers=octetHeader, ());
+  let blankConfig = Axios.makeConfig(~auth, ());
+
+  let myCIDs = () =>
```

Comment:
Definitions are right inline, and because the auth and baseURL are in scope we don't need partially-applied functions.